python/mxnet/contrib/quantization.py

```diff
@@ -286,12 +283,12 @@ def _get_optimal_threshold(arr, num_bins=8001, num_quantized_bins=255):
     min_val = np.min(arr)
     max_val = np.max(arr)
     th = max(abs(min_val), abs(max_val))
+    if min_val >= 0:
+        num_quantized_bins = (num_quantized_bins // 2) * 4 + 1
```

Comment:
For the case min_val > 0, unit8 should be used for quantized data, thus we should double the num_quantized_bins to make it suitable of unit8 range. This is the reason that entropy may produce even worse result than naive method, which is actually using 7bit in uint8.

Reply:
If I understand this correctly, you want to have 255 bins for positive values, 255 bins for negative values, and one bin for the zero value, although non-positive bins should not hold any values. Shouldn't this require 511 bins, instead of 255 // 2 * 4 +1 = 509 bins?

Reply:
Your understand is correct. Should change to `num_quantized_bins = num_quantized_bins * 2 + 1`.  But I removed this from this PR as it may affact GPU accuracy.